### PR TITLE
Enhancement/IMA-2-Main-Net connectivity improvements in beta

### PR DESCRIPTION
**Pull request by sergiy-skalelabs**
- IMA-2-Main-Net connectivity improvements

### CODE REVIEW

File: libskutils/src/rest_call.cpp [hunk 1]
```diff
@@ -78,6 +96,241 @@ std::string data_t::extract_json_id() const {
...
+        } catch ( std::exception& e ) {
+            throw std::runtime_error( "sz_cli: could not read client certificate file \"" +
+                                      optsSSL.client_cert + "\", exception is: " + e.what() );
+        }
+        try {
+            key_ = stat_f2s( optsSSL.client_key );
+            if ( key_.empty() )
+                throw std::runtime_error( "loaded data is empty" );
+        } catch ( std::exception& e ) {
+            throw std::runtime_error( "sz_cli: could not client key read file \"" +
```
Comment (cstrangedk) [inline libskutils/src/rest_call.cpp:119]: please fix error message

File: libskutils/src/rest_call.cpp [hunk 2]
```diff
@@ -319,21 +605,28 @@ bool client::stat_auto_gen_json_id( nlohmann::json& jo ) {
 }
 
 data_t client::call( const nlohmann::json& joIn, bool isAutoGenJsonID, e_data_fetch_strategy edfs,
-    std::chrono::milliseconds wait_step, size_t cntSteps ) {
+    std::chrono::milliseconds wait_step, size_t cntSteps, bool isReturnErrorResponce ) {
```
Comment (cstrangedk) [inline libskutils/src/rest_call.cpp]: typo, please fix all instances. isReturnErrorResponce -> isReturnErrorResponse.

**Review by cstrangedk — CHANGES_REQUESTED** [review]

File: libskutils/src/rest_call.cpp (hunk 1 above)
Comment (sergiy-skalelabs) [reply libskutils/src/rest_call.cpp:119]: fixed

File: libskutils/src/rest_call.cpp (hunk 2 above)
Comment (sergiy-skalelabs) [reply libskutils/src/rest_call.cpp]: fixed, thanx

**Review by kladkogex — APPROVED** [review]

**Pull request merged**